### PR TITLE
Update the installation instructions.

### LINKED ISSUE #123: Binary installer links on the TLAPS website are broken
The binary installer links on the [TLAPS website](https://tla.msr-inria.inria.fr/tlaps/content/Home.html) for [Windows](https://tla.msr-inria.inria.fr/tlaps/content/Download/Binaries/Windows.html), [Linux](https://tla.msr-inria.inria.fr/tlaps/content/Download/Binaries/Linux.html) and [macOS](https://tla.msr-inria.inria.fr/tlaps/content/Download/Binaries/MacOS.html) are all broken at the moment.

### PR DESCRIPTION
**Pull request by kape1395**
Fixes #123.

### CODE REVIEW

**Comment by ahelwer** [comment]
Sure, I will test the installation docs tomorrow!

For uploading the docs to the INRIA server what do you mean exactly?

**Comment by kape1395** [comment]
> For uploading the docs to the INRIA server what do you mean exactly?

Docs from https://github.com/tlaplus/tlapm/tree/main/doc/web have to be published somehow at https://tla.msr-inria.inria.fr/tlaps/. That should be done in the CI (maybe on a release, not each commit). Such publishing is done for  `tlaplus` here: https://github.com/tlaplus/tlaplus/blob/03c7bf4f35ccf1217ac2c53e5bf60d1220c8b0d8/.github/workflows/main.yml#L180. Maybe TLAPM CI scripts should be updated similarly.

**Comment by ahelwer** [comment]
@kape1395 the docs website is now served using github pages from the main branch of this repo at http://proofs.tlapl.us/doc/web/content/Home.html

**Comment by kape1395** [comment]
@ahelwer, thanks for the update! Maybe you have reviewed the changes to the installation instructions?

@lemmy, we should redirect somehow https://tla.msr-inria.inria.fr/tlaps/ to https://proofs.tlapl.us/, otherwise it will be confusing to have two sites for docs. Only the first one is suggested by google, for me at least.

**Comment by lemmy** [comment]
> @lemmy, we should redirect somehow https://tla.msr-inria.inria.fr/tlaps/ to https://proofs.tlapl.us/, otherwise it will be confusing to have two sites for docs. Only the first one is suggested by google, for me at least.

Done:

```shell
root@tla:/etc/apache2/sites-enabled# tail -7 default-ssl.conf

	Redirect 301 /tlaps https://proofs.tlapl.us/

	</VirtualHost>
</IfModule>
```

**Comment by kape1395** [comment]
@lemmy, this works partially. Paths are different, redirects should consider this, IMO. For example Google suggests <https://tla.msr-inria.inria.fr/tlaps/content/Download/Binaries.html>

Which is redirected to <https://proofs.tlapl.us//content/Download/Binaries.html> and gets 404.
It should be redirected to <https://proofs.tlapl.us/doc/web/content/Download/Binaries.html> (consider `doc/web` in the path).

**Comment by lemmy** [comment]
Do you know how to set this up in Apache 2?

**Comment by lemmy** [comment]
Please try again and let me know:

```shell
root@tla:/etc/apache2/sites-enabled# tail -7 default-ssl.conf

	Redirect 301 /tlaps https://proofs.tlapl.us/doc/web

	</VirtualHost>
</IfModule>
```

**Comment by kape1395** [comment]
Looks good now. Thanks!

File: doc/web/content/Download/Source.html
Comment (ahelwer) [inline doc/web/content/Download/Source.html:24]: Do we want to remove mention of Cygwin here and focus on WSL support?

File: doc/web/content/Download/Binaries/Linux.html
Comment (ahelwer) [inline doc/web/content/Download/Binaries/Linux.html:24]: In the (near) future when we switch to a rolling release model I suppose we will want to use the /latest/ tag along with a link to a binary that does not have the version in the name.

**Review by ahelwer — APPROVED** [review]

File: doc/web/content/Download/Source.html
Comment (kape1395) [reply doc/web/content/Download/Source.html]: Thanks, I removed the mention of Cygwin.

File: doc/web/content/Download/Binaries/Linux.html
Comment (kape1395) [reply doc/web/content/Download/Binaries/Linux.html:24]: I agree. This will have to be changed in the future.
For now, I just wanted to keep links valid after a new release is made.

**Comment by kape1395** [comment]
@lemmy, @ahelwer, the MacOS version cannot be built anymore.
The scripts refer to macos-latest, and it points now to `macos-14-arm64`. It was `macos-12` in the latest successful build.

The old Isabelle (2011) does not work with the ARM chips.
The new Isabelle (2024) is in progress. All the non-skipped proofs in the Examples repo have been built successfully, but some degradations still exist (it looks like something with integers/naturals).

The question:
  - Should I merge this PR and have the docs updated (but the build on the main is broken).
  - Wait until the Isabelle branch is merged (#109, #124) and have the Installation instructions with broken links until then.

**Comment by ahelwer** [comment]
Can we split the macOS build into intel and arm64?

**Comment by kape1395** [comment]
I think we will have to. Maybe for now, we can stick to `macos-13` for the Intel version?
See https://github.com/actions/runner-images?tab=readme-ov-file#available-images

**Comment by ahelwer** [comment]
Is there a way to do cross-compilation instead of having to run the build on an intel mac?

**Comment by kape1395** [comment]
I don't know; I never tried to do that.

**Comment by lemmy** [comment]
Are we forced to compile on `macos-latest`? Why not an older runner: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories?

**Comment by kape1395** [comment]
It works with `macos-13`, so I'll keep it like this for now.

**Pull request merged**
